railties/test/application/rake/multi_dbs_test.rb

```diff
@@ -112,6 +112,83 @@ def db_migrate_and_schema_dump_and_load(format)
         end
       end
 
+      def db_migrate_and_schema_dump_and_load_one_database(format, database)
+        Dir.chdir(app_path) do
+          generate_models_for_animals
+          rails "db:migrate:#{database}", "db:#{format}:dump:#{database}"
+
+          if format == "schema"
+            if database == "primary"
+              schema_dump = File.read("db/#{format}.rb")
+              assert_not(File.exist?("db/animals_#{format}.rb"))
+              assert_match(/create_table \"books\"/, schema_dump)
+            else
+              assert_not(File.exist?("db/#{format}.rb"))
+              schema_dump_animals = File.read("db/animals_#{format}.rb")
+              assert_match(/create_table \"dogs\"/, schema_dump_animals)
+            end
+          else
+            if database == "primary"
+              schema_dump = File.read("db/#{format}.sql")
+              assert_not(File.exist?("db/animals_#{format}.sql"))
+              assert_match(/CREATE TABLE (?:IF NOT EXISTS )?\"books\"/, schema_dump)
+            else
+              assert_not(File.exist?("db/#{format}.sql"))
+              schema_dump_animals = File.read("db/animals_#{format}.sql")
+              assert_match(/CREATE TABLE (?:IF NOT EXISTS )?\"dogs\"/, schema_dump_animals)
+            end
+          end
+
+          rails "db:#{format}:load:#{database}"
+
+          ar_tables = lambda { rails("runner", "p ActiveRecord::Base.connection.tables").strip }
+          animals_tables = lambda { rails("runner", "p AnimalsBase.connection.tables").strip }
+
+          if database == "primary"
+            assert_equal '["schema_migrations", "ar_internal_metadata", "books"]', ar_tables[]
+            assert_equal '[]', animals_tables[]
+          else
+            assert_equal '[]', ar_tables[]
+            assert_equal '["schema_migrations", "ar_internal_metadata", "dogs"]', animals_tables[]
+          end
+        end
+      end
+
+      def db_test_prepare_spec_name(spec_name, schema_format)
+        Dir.chdir(app_path) do
+          app_file "config/initializers/enable_sql_schema_format.rb", <<~RUBY
+            Rails.application.config.active_record.schema_format = :#{schema_format}
+          RUBY
```

Comment:
This doesn't seem to be working as-written. You'll notice this causes the `test "db:test:prepare:spec_name works for the **** database with a sql schema"` tests to fail. 

Any guidance here would be very much appreciated!

Reply:
https://github.com/rails/rails/commit/4ff8e012d689ecafe3c7f777da6685cfa32ea40c